### DISCUSSION TITLE
Insert images, call via keyboard shortcut, collapse hints

### DISCUSSION THREAD

**Discussion by enshtein** [discoverlance-com/filament-page-hints#10] [Ideas]
Hello.

First of all, thanks a lot for the plugin!

Here are some ideas to make your plugin even better:

- the ability to insert images in the text, without loading directly from the buffer via <img src="data:image/
- the ability to **collapse** and **expand** blocks of hints
- make it possible to assign a keyboard shortcut to call the hint panel - for example: **Ctrl + F1**
- there are no options in **config/filament-page-hints.php**: should_register_navigation, navigation_sort, navigation_badge, navigation_group

**Comment by enshtein** [comment]
As for inserting images, in general I see that the task is solved through the editor parameters (config/filament-page-hints.php):

```
    'toolbar_buttons' => [
        ...
        'attachFiles',
    ],
```

**Comment by discoverlance-com** [comment]
I think we can at the moment do with the `'attachFiles'` option in the config for images. Or do you have another idea on how what you suggested could be different from that?

Thanks for your suggestions. Yes, I think the other ideas are very good and so will look to add those in the next release, thus,

1. the ability to collapse and expand blocks of hints
2. make it possible to assign a keyboard shortcut to call the hint panel - for example Ctrl + F1
3. there are no options in config/filament-page-hints.php: should_register_navigation, navigation_sort, navigation_badge, navigation_group

I will release a patch by Monday with the config options and then look to add points 1 and 2.
